python/lsst/the/monster/assemble.py

```diff
@@ -0,0 +1,167 @@
+import esutil
+import os
+import numpy as np
+import lsst.utils
+
+from .splinecolorterms import ColortermSpline
+from .refcats import GaiaXPInfo, GaiaDR3Info, SkyMapperInfo, PS1Info, VSTInfo, DESInfo, SynthLSSTInfo
+from .utils import read_stars, makeMonsterSchema, makeMonsterCat
+
+__all__ = ["AssembleMonsterRefcat"]
+
+"""
+This Python script starts with the full Gaia DR3 catalog, then assembles a
+monster refcat by doing the following:
+
+For each shard:
+1. Read the full Gaia DR3 catalog for the shard
+2. Initialize 18 columns (ugrizy fluxes and their errors, and a source flag
+    (integer)) for the results
+(3-6): Within a loop over surveys in order from lowest to highest priority:
+3. Read each of the (already transformed to the DES system) refcats for the
+    shard
+4. Transform each refcat to the (synthetic) LSST system
+5. Match each refcat to the Gaia DR3 catalog
+6. Update the fluxes, flux errors, and flags whenever a value is non-NaN
+"""
+
+
+class AssembleMonsterRefcat:
+    def __init__(self,
+                 gaia_reference_class=GaiaDR3Info,
+                 catalog_info_class_list=[VSTInfo, SkyMapperInfo,
+                                          PS1Info, GaiaXPInfo, DESInfo],
+                 write_path_inp=None,
+                 testing_mode=False,
+                 ):
+
+        self.gaia_reference_info = gaia_reference_class()
+        self.catalog_info_class_list = [cat_info() for cat_info
+                                        in catalog_info_class_list]
+        self.synth_lsst_info = SynthLSSTInfo()
```

Comment:
See next comment about latiss (typing these in wrong order), but this will need to be configurable to do not just synth lsst.  (And you'll run into the problem during testing that this will need to be configurable).

Reply:
I accomplished it by doing this, but it's only configurable to include things that are in the dict:
`                 synth_system='LSST',
                 ):

        self.gaia_reference_info = gaia_reference_class()
        self.catalog_info_class_list = [cat_info() for cat_info
                                        in catalog_info_class_list]
        synth_info_dict = {'LSST': SynthLSSTInfo}
        self.synth_info = synth_info_dict[synth_system]()
`